### PR TITLE
Refactor Section::RemoveService

### PR DESCRIPTION
**Pull request by chrisroos**
I don't think these single line methods were adding anything.

### CODE REVIEW

File: app/services/section/remove_service.rb [hunk 1]
```diff
@@ -9,18 +9,29 @@ def initialize(user:, manual_id:, section_uuid:, attributes:)
   end
 
   def call
+    begin
+      manual = Manual.find(manual_id, user)
+    rescue KeyError
+      raise ManualNotFoundError.new(manual_id)
```
Comment (floehopper) [inline app/services/section/remove_service.rb]: Possibly out-of-scope: I don't think `Manual.find` raises a `KeyError` any more; I think it raises a `Manual::NotFoundError` instead. Thus we could probably ditch this `rescue` block and rescue the latter exception in `ManualWithdrawer#execute`.

File: app/services/section/remove_service.rb [hunk 2]
```diff
@@ -9,18 +9,29 @@ def initialize(user:, manual_id:, section_uuid:, attributes:)
...
+    end
+
+    section = manual.sections.find { |s| s.uuid == section_uuid }
```
Comment (floehopper) [inline app/services/section/remove_service.rb]: Minor: See my [earlier comment](https://github.com/alphagov/manuals-publisher/pull/1134#discussion_r118053067) about extracting a `Manual#find_section` method.

File: app/services/section/remove_service.rb (hunk 1 above)
Comment (chrisroos) [inline app/services/section/remove_service.rb]: Ah, that's useful to know! I'm going to leave this for now as I can imagine addressing it as I continue to work through these services ensuring that the save to the publishing api happens before the save to our database.

File: app/services/section/remove_service.rb (hunk 2 above)
Comment (chrisroos) [inline app/services/section/remove_service.rb]: I'm planning to introduce a `Manual#find_section` method outside of this PR.

**Comment by chrisroos** [comment]
Thanks @floehopper. I've rebased on master and force pushed in preparation for merging.

**Pull request merged**